app/styles/ui/_side-by-side-diff.scss

```
@@ -326,9 +326,27 @@
         left: 100px;
       }
 
-      &.hunk-info .line-number {
-        background: var(--diff-hunk-gutter-background-color);
-        border-color: var(--diff-hunk-border-color);
+      &.hunk-info {
+        .line-number {
+          background: var(--diff-hunk-gutter-background-color);
+          border-color: var(--diff-hunk-border-color);
+        }
+        .hunk-expansion-handle {
+          background: var(--diff-hunk-gutter-background-color);
+          border-right-width: 1px;
+          border-right-style: solid;
+          border-color: var(--diff-hunk-gutter-background-color);
+          &.selectable:hover {
+            border-color: var(--diff-hover-background-color);
+          }
+          &:not(.selectable) span {
+            // This must be the height of an octicon, so that when a hunk handle
+            // of any type is replaced by an "empty" hunk handle (e.g. when a
+            // file is completely expanded), the height of the whole line
+            // remains the same.
+            height: 16px;
+          }
+        }
```

```suggestion
        .hunk-expansion-handle {
          background: var(--diff-hunk-gutter-background-color);
          border-right-width: 1px;
          border-right-style: solid;
          border-color: var(--diff-hunk-gutter-background-color);
          align-self: stretch;
          align-items: center;
          
          &.selectable:hover {
            border-color: var(--diff-hover-background-color);
          }
        }
```

I think this will do what you need... I tested in the case of an expanded file and not expanded file and I think it behaves correctly.

---

The `align-self: stretch` makes it take up it's parents height  and then the `align-items: center` centers the span inside like div's parent was doing to this div to achieve centering.

---

(And I think this works because the parent is a flex box itself... but I am a little fuzzy on that).

---

Thank you!! It's not perfect as the CodeMirror behavior (it still gets a bit shorter), but it's the same behavior displayed in the split diff. We can address that "glitchy" behavior in a separate PR.